pyacq/core/tools.py

```diff
@@ -22,10 +22,13 @@ class ThreadPollInput(QtCore.QThread):
     The `process_data()` method may be reimplemented to define other behaviors.
     """
     new_data = QtCore.Signal(int,object)
-    def __init__(self, input_stream, timeout=200, parent=None):
+    def __init__(self, input_stream, timeout=200, return_data=None, parent=None):
         QtCore.QThread.__init__(self, parent)
         self.input_stream = weakref.ref(input_stream)
         self.timeout = timeout
+        self.return_data = return_data
+        if self.return_data is None:
+            self.return_data = self.input_stream()._own_buffer
```

Comment:
I think my main idea is maximum reduce job at pooling level.

A very commun scenario is:
 * one sender send big buffer very often (10ms)
 * the stream is shearredmem
 * many receivers (one by channel) pool the index and are waiting to have enougth data in the ringbuffer before doing something at a slower rate (200ms).

In that case we must not construct any arrays or python object every 10ms for many receiver when pooling even if the object is small because we do not need then and the overhead could be consequent.

So:
 * yes for having a commun API for all stream classes
 * I prefer to have complex API that fit all stream scenario.
 * I prefer to keep this API with some extra kargs and not introduce new object for defering something when we do not need.

In streams module, performences are more important than simplification. No ?

I plan to introduce after this PR an OpenCLBufferStream that could even make the API with more kargs.
We could also think of having a UdpSocketStream  or something like that.
In this new scenarios pooling and having the buffer have also sligthly different constraint mining than this 2 streams.